### PR TITLE
Fixed #24743 -- Grouped consecutive migrations in effective migration plan

### PR DESCRIPTION
**Pull request by MarkusH**
According to a [comment on Trac](https://code.djangoproject.com/ticket/24743#comment:14) this PR speeds up the ModelState rendering by about 90%.

Ticket: https://code.djangoproject.com/ticket/24743

### CODE REVIEW

File: django/db/migrations/executor.py [hunk 1]
```diff
@@ -66,50 +68,97 @@ def migrate(self, targets, plan=None, fake=False, fake_initial=False):
...
+                    groups.append(current_group)
+                    current_group = OrderedDict()
+                    current_group[migration] = (0, backwards)
+                    compute_state_for.add(migration)
+                    current_index = 1
+            last = (migration, backwards)
+        groups.append(current_group)
+
+        # Phase 2 -- Compute and store project states. The migration that first
+        # whose "before-state" is stored also triggers the rendering of
```
Comment (charettes) [inline django/db/migrations/executor.py]: that first whose?

**Comment by dacalation** [comment]
As a heads up, I tried this patch on my app and didn't see any change in time to apply an empty migration.  We have 70 migrations with 155 models and the full operation takes 6 minutes.

**Comment by davidcrawford** [comment]
As a heads up, I tried this patch on my app and didn't see any change in time to apply an empty migration. We have 70 migrations with 155 models and the full operation takes 6 minutes.

**Comment by MarkusH** [comment]
Thanks @davidcrawford. It is not about applying the migrations themselves, it's about the 'rendering model state' step before. You can get a timing output when you run `manage.py migrate` with `-v 3`.

**Comment by davidcrawford** [comment]
```
Running migrations:
  Rendering model states... DONE (321.175s)
  Applying rosemeta.0075_dummy... OK (0.052s)
```

**Comment by MarkusH** [comment]
Since #5178 yields far better results in all my tests I go with that implementation.

**Pull request closed without merging**